### DISCUSSION TITLE
Add the ability to choose what audio device Chatterino uses to play sounds

### DISCUSSION THREAD

**Discussion by Toxijuice** [Chatterino/chatterino2#5590] [Ideas]
Currently, Chatterino appears to only use the default audio device to play sounds. It would be beneficial to have a setting to manually choose which audio device Chatterino uses, instead.

In my case, I use Chatterino while streaming to play a sound for specific messages so I don't miss them. My default audio device is what the stream hears, but I'd rather not have the stream hear Chatterino sounds.

My workaround is to use [EarTrumpet](https://github.com/File-New-Project/EarTrumpet) to switch the audio device to one that only I can hear, but I have to do this every time I start up Chatterino. Furthermore, I cannot switch to another audio device until at least one highlight sound has already played. _And_ there doesn't seem to be a way to play a sound effect for your own messages, so I have to use a bot command to trigger the sound effect.

It's not the worst thing in the world, but it's very cumbersome to do every time I use Chatterino, and would be completely solved if I had the option to select a different audio device.

Also, this is likely very similar to what was suggested in #3240, but that seemed more akin to a bug report potentially outside the domain of Chatterino. Windows does "remember" the device set, but Chatterino likely does not check Windows defaults when initializing its audio playback system.

**Comment by pajlada** [comment]
On Linux, I solve this by telling my Sound Manager that Chatterino should use a separate output. My Sound Manager remembers this after reboot.

On Windows, it looks like it should be possible to do the same with System > Sound > Volume mixer, and selecting the Output device for Chatterino. For me this works on Windows 11 using Chatterino v2.5.1

If this doesn't work, can you share what Chatterino version & Windows version you're on?

**Reply by Toxijuice** [reply]
> If this doesn't work, can you share what Chatterino version & Windows version you're on?

I'm on Windows 10, Chatterino 2.5.1.

Windows 11 might do things a bit differently, but the equivalent on Windows 10 is System > Sound > App volume and device preferences. As far as I understand EarTrumpet changes these same exact values, just from a different UI. So it has all the same issues I already face, as they're functionally identical.

Interestingly, I've had instances where Windows reports that Chatterino is using the audio device I want it to, when it definitely was using the default audio device instead.

So maybe a Windows 10 bug?

**Reply by pajlada** [reply]
Could be, I don't have a Windows 10 system available to test. Once you inevitably upgrade, and the problem is still happening, I'd be happy to look more into it.
